jobs/cache_maintenance/tests/test_clean_cache.py

```diff
@@ -0,0 +1,71 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2023 The HuggingFace Authors.
+
+import os
+from http import HTTPStatus
+from pathlib import Path
+from unittest.mock import patch
+
+from huggingface_hub.hf_api import DatasetInfo
+from libcommon.simple_cache import has_some_cache, upsert_response
+
+from cache_maintenance.cache_cleaner import clean_cache
+
+
+def test_clean_cache() -> None:
+    assets_directory = "/tmp/assets"
+    cached_assets_directory = "/tmp/cached-assets"
+    dataset = "dataset"
+    os.makedirs(f"{assets_directory}/{dataset}", exist_ok=True)
+    os.makedirs(f"{cached_assets_directory}/{dataset}", exist_ok=True)
+
+    asset_file = Path(f"{assets_directory}/{dataset}/image.jpg")
+    asset_file.touch()
+    assert asset_file.is_file()
+
+    cached_asset_file = Path(f"{cached_assets_directory}/{dataset}/image.jpg")
+    cached_asset_file.touch()
+    assert cached_asset_file.is_file()
+
+    upsert_response(
+        kind="dataset-config-names",
+        dataset=dataset,
+        content={"config_names": [{"dataset": dataset, "config": "config"}]},
+        http_status=HTTPStatus.OK,
+    )
+
+    upsert_response(
+        kind="config-split-names-from-streaming",
+        dataset=dataset,
+        config="config",
+        content={"splits": [{"dataset": dataset, "config": "config", "split": "split"}]},
+        http_status=HTTPStatus.OK,
+    )
+    assert has_some_cache(dataset=dataset)
+
+    dataset_info = DatasetInfo(id=dataset)
+    with patch("cache_maintenance.cache_cleaner.get_supported_dataset_infos", return_value=[dataset_info]):
+        clean_cache(
+            hf_endpoint="hf_endpoint",
+            hf_token="hf_token",
+            assets_directory=assets_directory,
+            cached_assets_directory=cached_assets_directory,
+        )
+
+    assert asset_file.is_file()
+    assert cached_asset_file.is_file()
+    assert has_some_cache(dataset=dataset)
+
+    with patch("cache_maintenance.cache_cleaner.get_supported_dataset_infos", return_value=[]):
```

Comment:
This part frightens me: if the code does the same, for some reason, the following part returns an empty list, we will delete everything:

https://github.com/huggingface/datasets-server/blob/cbd38812d1d08da24f10ce6501e4c520687b2faf/libs/libcommon/src/libcommon/dataset.py#L172

We could add a safeguard: cancel the action if we get an empty list. Even a bit safer: cancel if the list of datasets is under a threshold (`20,000`, for example). Imagine an issue with the pagination that would return only 10,000 results.

Reply:
Sure, I added a validation on https://github.com/huggingface/datasets-server/pull/1822/commits/819c0cd20e8b99df071842f5f8d2047a6d5832a8